### PR TITLE
fix(33377): completions for string literals don't follow quoteStyle preference

### LINKED ISSUE #31507: Quote preference not used for literal type completions
* use single quote preference
* have the snippet below, trigger intellisense
* :bug: double quotes are used

```ts
type Foo = 'abc' | 'def';

let f: Foo = 
```

Screenshot 2019-05-21 at 12 16 03

### LINKED ISSUE #33377: completions for string literals don't follow quoteStyle preference
*Template Info added by @mjbvz*

**TS Version**: 3.7.0-dev.20190910

**Search terms**
- intellisense
- suggest, suggestions
- quote
- quoteStyle

*Original report*

---

Steps to Reproduce:

1. In settings, set : "typescript.preferences.quoteStyle": "single"
2. Create a file like this : 

````typescript
export enum E {
    VALID, INVALID
}

export interface a {
    test: keyof typeof E;
}

const a: a = {
    // should suggest with simple quote
    test: 
}
````

3. Focus after 'test:' and launch suggestions with <kbd>Ctrl</kbd>+<kbd>Space</kbd>
4. Choose first

Does this issue occur when all extensions are disabled?: Yes

Is there a way to autocomplete this with single quotes ? I haven't found any settings of this kind, except quoteStyle preferences.
Paramètres - Visual Studio Code 11_09_2019 15_49_24

### PR DESCRIPTION
**Pull request by a-tarasyuk**
Fixes #33377
Fixes #31507

### CODE REVIEW

File: src/services/completions.ts [hunk 1]
```diff
@@ -317,10 +317,20 @@ namespace ts.Completions {
...
+    function completionNameForLiteral(literal: string | number | PseudoBigInt, preferences?: UserPreferences): string {
+        if (typeof literal === "object") {
+            return pseudoBigIntToString(literal) + "n";
+        }
+
+        if (isString(literal) && preferences) {
+            return quote(literal, preferences);
+        }
+
+        return JSON.stringify(literal);
```
Comment (sheetalkamat) [inline src/services/completions.ts]:
```suggestion
    return typeof literal === "object" ?
        pseudoBigIntToString(literal) + "n" :
        isString(literal) ?
            quote(literal, preferences) :
            JSON.stringify(literal);
```

File: src/services/completions.ts [hunk 2]
```diff
@@ -317,10 +317,20 @@ namespace ts.Completions {
         });
     }
 
-    const completionNameForLiteral = (literal: string | number | PseudoBigInt) =>
-        typeof literal === "object" ? pseudoBigIntToString(literal) + "n" : JSON.stringify(literal);
-    function createCompletionEntryForLiteral(literal: string | number | PseudoBigInt): CompletionEntry {
-        return { name: completionNameForLiteral(literal), kind: ScriptElementKind.string, kindModifiers: ScriptElementKindModifier.none, sortText: SortText.LocationPriority };
+    function completionNameForLiteral(literal: string | number | PseudoBigInt, preferences?: UserPreferences): string {
```
Comment (sheetalkamat) [inline src/services/completions.ts]: You want `preferences` to be passed from all the calling locations..

Comment (a-tarasyuk) [reply src/services/completions.ts]: @sheetalkamat Thanks for the feedback 👍

**Comment by sandersn** [comment]
@sheetalkamat this is ready to go, right?

**Comment by a-tarasyuk** [comment]
Seems this PR resolves two issues 

#33377
#31507

cc @sandersn @sheetalkamat @DanielRosenwasser

**Comment by sandersn** [comment]
(I edited the description to include the second bug)

**Pull request merged**